Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+
+                It 'Passes CPU Limit Test when same' {
+                  
+                    TestTargetResourceSame "cpuLimit" "add.cpu.limit" | Should be $true
+                }
+
+                It 'Fails CPU Limit Test Test when different' {
+                  
+                    TestTargetResourceDiff "cpuLimit" "add.cpu.limit" "1" "0" | Should be $false
+                }
+
+                It 'Passes CPU Action Test when same' {
+                  
+                    TestTargetResourceSame "cpuAction" "add.cpu.action" | Should be $true
+                }
+
+                It 'Fails CPU Action Test when different' {
+                  
+                    TestTargetResourceDiff "cpuAction" "add.cpu.action" "Throttle" "NoAction" | Should be $false
+                }
+
+                It 'Passes CPU Reset Interval Test when same' {
+                  
+                    TestTargetResourceSame "cpuResetInterval" "add.cpu.resetInterval" | Should be $true
+                }
+
+                It 'Fails CPU Reset Interval Test when different' {
+                  
+                    TestTargetResourceDiff "cpuResetInterval" "add.cpu.resetInterval" "00:15:00" "00:05:00" | Should be $false
+                }
+
+                It 'Passes CPU Smp Affinitized Test when same' {
+                  
+                    TestTargetResourceSame "cpuSmpAffinitized" "add.cpu.smpAffinitized" | Should be $true
+                }
+
+                It 'Fails CPU Smp Affinitized Test when different' {
+                  
+                    TestTargetResourceDiff "cpuSmpAffinitized" "add.cpu.smpAffinitized" "true" "false" | Should be $false
+                }
+
+                It 'Passes CPU Smp Processor Affinity Mask Test when same' {
+                  
+                    TestTargetResourceSame "cpuSmpProcessorAffinityMask" "add.cpu.smpProcessorAffinityMask" | Should be $true
+                }
+
+                It 'Fails CPU Smp Processor Affinity Mask Test when different' {
+                  
+                    TestTargetResourceDiff "cpuSmpProcessorAffinityMask" "add.cpu.smpProcessorAffinityMask" "4294967294" "4294967295" | Should be $false
+                }
+
+                It 'Passes CPU Smp Processor Affinity Mask 2 Test when same' {
+                  
+                    TestTargetResourceSame "cpuSmpProcessorAffinityMask2" "add.cpu.smpProcessorAffinityMask2" | Should be $true
+                }
+
+                It 'Fails CPU Smp Processor Affinity Mask 2 Test when different' {
+                  
+                    TestTargetResourceDiff "cpuSmpProcessorAffinityMask2" "add.cpu.smpProcessorAffinityMask2" "4294967294" "4294967295" | Should be $false
+                }
+            }
+            
+<#            Context 'Set-TargetResource' {
+                It 'Should set autoStart Test ' {
+                    
+                    SetTargetResource "autoStart" "add.autoStart" "true" "false"| Should be $true
```

Comment:
I'd recommend these tests become

````powershell
Mock Invoke-AppCmd {}
Mock Invoke-AppCmd {return $global:PoolCfg} -ParameterFilter {$Arguments.Contains('list')}

It 'Should set autoStart Test ' {
    Set-TargetResource -Name 'DefaultAppPool' -AutoStart 'False'
    Assert-MockCalled Invoke-AppCmd -ParameterFilter {$Arguments.Contains('set')} -Times 1 -Exactly
    Assert-MockCalled Invoke-AppCmd -ParameterFilter {$Arguments.Contains('/autoStart:False')} -Times 1 -Exactly
}
````

This way we can test a couple of things.
1) Invoke-AppCmd -Arguments set is only called the one time we expect it to be (given the current logic we expect set to only be called on the one difference.
2) Invoke-AppCmd is calling the specific change we expect (in this case autostart)